dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1389,152 +1389,201 @@ public Identifier getRelatedIdentifier(Contentlet contentlet,String relationship
+                        respectFrontendRoles);
+
+        if (response.getHits() == null) {
+            return result;
+        }
+
+        for (SearchHit sh : response.getHits()) {
+            try {
+                Map<String, Object> sourceMap = sh.getSourceAsMap();
+                if (sourceMap.get(relationshipName) != null) {
+                    ((ArrayList<Map>) sourceMap.get(relationshipName)).stream().forEach(child -> {
+                        try {
+                            result.add(findContentletByIdentifierAnyLanguage(
+                                    (String) child.get("identifier")));
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                }
+            } catch (Exception e) {
+                Logger.error(this, e.getMessage(), e);
             }
         }
-        return q;
+
+        return result;
     }
 
-    private String getRelatedContentESQuery(Contentlet contentlet,Relationship rel, boolean pullByParent) {
-        final boolean isSameStructureRelationship = FactoryLocator.getRelationshipFactory()
-                .sameParentAndChild(rel);
-        String q;
+    private List<Contentlet> getRelatedParents(final Contentlet contentlet, final Relationship rel,
+            final User user, final boolean respectFrontendRoles)
+            throws DotSecurityException, DotDataException {
+        final List<Contentlet> result = new ArrayList<>();
+        final String relationshipName = rel.getRelationTypeValue().toLowerCase();
 
-        if(isSameStructureRelationship) {
-            String disc = pullByParent?"-parent":"-child";
-            q = "+type:content +" + rel.getRelationTypeValue() + disc + ":" + contentlet.getIdentifier();
-            if(!InodeUtils.isSet(contentlet.getIdentifier()))
-                q = "+type:content +" + rel.getRelationTypeValue() + disc + ":" + "0";
+        SearchResponse response = APILocator.getEsSearchAPI()
+                .esSearchRelated(contentlet.getIdentifier(), relationshipName, true,false, user,
+                        respectFrontendRoles);
 
-        } else {
-            q = "+type:content +" + rel.getRelationTypeValue() + ":" + contentlet.getIdentifier();
-            if(!InodeUtils.isSet(contentlet.getIdentifier()))
-                q = "+type:content +" + rel.getRelationTypeValue() + ":" + "0";
+        if (response.getHits() == null) {
+            return result;
         }
 
-        return q;
-    }
+        for (SearchHit sh : response.getHits()) {
+            try {
+                Map<String, Object> sourceMap = sh.getSourceAsMap();
+                if (sourceMap.get("identifier") != null) {
+                    result.add(findContentletByIdentifierAnyLanguage(sourceMap.get("identifier").toString()));
+                }
+            } catch (Exception e) {
+                Logger.error(this, e.getMessage(), e);
```

Comment:
Please log the content's Identifier and relationship name too.

Reply:
The catch was not necessary, so I removed it